src/composables/permissions.ts

```diff
@@ -0,0 +1,138 @@
+import { METHODS } from '@aeternity/aepp-sdk';
+import type { IPermission, PermissionRegistry } from '@/types';
+import { STORAGE_KEYS } from '@/constants';
+import { aettosToAe } from '@/protocols/aeternity/helpers';
+import migratePermissionsVuexToComposable from '@/migrations/003-permissions-vuex-to-composable';
+import { useStorageRef } from './composablesHelpers';
+
+interface ITransactionSignPermissionOptions {
+  amount?: number;
+  fee?: number;
+  nameFee?: number;
+}
+
+const permissions = useStorageRef<PermissionRegistry>(
+  {},
+  STORAGE_KEYS.permissions,
+  {
+    backgroundSync: true,
+    migrations: [
+      migratePermissionsVuexToComposable,
+    ],
+  },
+);
+
+export function usePermissions() {
+  function setTransactionSignLimitLeft(host: string, value: number) {
+    permissions.value[host] = {
+      ...permissions.value[host],
+      transactionSignLimitLeft: value,
+    };
+  }
+
+  function resetTransactionSignLimitLeft(host: string) {
+    permissions.value[host] = {
+      ...permissions.value[host],
+      transactionSignLimitLeft: permissions.value[host].transactionSignLimit,
+      transactionSignFirstAskedOn: (new Date()).toISOString(),
+    };
+  }
+
+  function addPermission(permission: IPermission) {
+    permissions.value[permission.host] = permission;
+  }
+
+  function removePermission(host: string) {
+    delete permissions.value[host];
+  }
+
+  function checkTransactionSignPermission(
+    host: string,
+    { amount = 0, fee = 0, nameFee = 0 }: ITransactionSignPermissionOptions,
+  ): boolean {
+    const {
+      transactionSignLimit,
+      transactionSignLimitLeft,
+      transactionSignFirstAskedOn,
+    } = permissions.value[host] || {};
+    const limitLeft = transactionSignLimitLeft - +aettosToAe(+amount + +fee + +nameFee);
+
+    if (!transactionSignLimit) {
+      return false;
+    }
+    if (
+      !transactionSignFirstAskedOn
+      || (
+        new Date().getTime() - new Date(transactionSignFirstAskedOn).getTime()
+        >= 24 * 60 * 60 * 1000
```

Comment:
What do you think of using dayjs for date and time comparison?

Reply:
After doing some research I think there is a chance we could remove the `dayjs` lib as it gives no real benefit. We do only a little of date transformations and comparisons and having a 3rd party lib for this makes no sense for me.

Reply:
I think the most important usage of it is when we display relativeTime in NotificationList & TransactionList - that's when it was introduced.